libraries/shared/storage/diff_repository.go

```diff
@@ -62,37 +61,10 @@ func (repository diffRepository) CreateBackFilledStorageValue(rawDiff types.RawD
 	return err
 }
 
-func (repository diffRepository) GetNewDiffs(diffs chan types.PersistedDiff, errs chan error, done chan bool) {
-	rows, queryErr := repository.db.Queryx(`SELECT * FROM public.storage_diff WHERE checked IS false`)
-	if queryErr != nil {
-		logrus.Errorf("error getting unchecked storage diffs: %s", queryErr.Error())
-		if rows != nil {
-			closeErr := rows.Close()
-			if closeErr != nil {
-				logrus.Errorf("error closing rows: %s", closeErr.Error())
-			}
-		}
-		errs <- queryErr
-		return
-	}
-
-	if rows != nil {
-		for rows.Next() {
-			var diff types.PersistedDiff
-			scanErr := rows.StructScan(&diff)
-			if scanErr != nil {
-				logrus.Errorf("error scanning diff: %s", scanErr.Error())
-				closeErr := rows.Close()
-				if closeErr != nil {
-					logrus.Errorf("error closing rows: %s", closeErr.Error())
-				}
-				errs <- scanErr
-			}
-			diffs <- diff
-		}
-	}
-
-	done <- true
+func (repository diffRepository) GetNewDiffs() ([]types.PersistedDiff, error) {
+	var result []types.PersistedDiff
+	err := repository.db.Select(&result, `SELECT * FROM public.storage_diff WHERE checked IS false LIMIT 500`)
```

Comment:
super small, and doesn't necessarily have to be done in this PR, but I wonder if passing this limit in as a config would be good in case we want to tweak it in the future.

Reply:
I think maybe you're looking at an older commit - this is controlled via an arg passed to `GetNewDiffs` now from the watcher. Though maybe it still makes sense to add to config so we don't need to change the code to change the limit? 🤔

Reply:
I'm leaning toward leaving it in the code for now so that we don't need to update the env vars sent to the container on a redeploy (or synchronize a config change in vdb-mcd-transformers config files before merging any PRs over there) but would be down to add a new story for this if we want to avoid updating the code to change it

Reply:
whoops! yep, was looking only at the first commit. 🤦‍♀️That makes sense to leave it in code for now!